qontract_utils/qontract_utils/kubernetes/client.py

```
@@ -0,0 +1,328 @@
+        except ApiError as e:
+            if e.status.code == _HTTP_NOT_FOUND:
+                return
+            raise from_api_error(e) from e
+
+    def close(self) -> None:
+        """Release the underlying lightkube client and httpx connection pool.
+
+        Workaround: lightkube has no public close() on the sync client.
+        https://github.com/gtsystem/lightkube/issues/144
+        """
+        self._client._client._client.close()  # type: ignore[attr-defined]
```

Ruff (0.15.20)

[error] Private member accessed: `_client`

(SLF001)